projects/ad9361/src/platform_xilinx/platform.c

```diff
@@ -79,111 +79,118 @@ XSpi			spi_instance;
 XGpio_Config	*gpio_config;
 #endif
 
-/***************************************************************************//**
- * @brief spi_init
-*******************************************************************************/
-int32_t spi_init(uint32_t device_id,
-		 uint8_t  clk_pha,
-		 uint8_t  clk_pol)
+/**
+ * @brief Initialize the SPI communication peripheral.
+ * @param desc - The SPI descriptor.
+ * @param init_param - The structure that contains the SPI parameters.
+ * @return SUCCESS in case of success, FAILURE otherwise.
+ */
+int32_t spi_init(struct spi_desc **desc,
+		 const struct spi_init_param *param)
 {
+	spi_desc *descriptor;
+	int32_t ret;
 
-	uint32_t base_addr	 = 0;
-	uint32_t spi_options = 0;
-#ifdef _XPARAMETERS_PS_H_
+	descriptor = (struct spi_desc *) calloc(1, sizeof(*descriptor));
+	if (!descriptor)
+		return FAILURE;
 
-	spi_config = XSpiPs_LookupConfig(device_id);
-	base_addr = spi_config->BaseAddress;
+	descriptor->mode = param->mode;
+	descriptor->chip_select = param->chip_select;
+	descriptor->flags = param->flags;
```

Comment:
curious: Are all the members of the descriptor assigned a value in this function? Specifically type, id and max_speed_hz.

Reply:
Probably the other members, need an initial value.
The switch to the new driver is made in small steps. We switch to SPI now. 
"platform.c" will be replaced by "projects/drivers/xilinx_platform/platform_drivers.c", in the next pull request.
spi_init(); is a temporary copy from "platform_drivers.c". We can't use platform_drivers.c and platform.c at the same time, because of the conflicts between them.

Reply:
Got it. Thank you!